overlays/honeycomb.yml

```diff
@@ -0,0 +1,16 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:yaml", "yaml")
+#@ load("@ytt:data", "data")
+
+#@ def updates():
+#@overlay/match missing_ok=True
+honeycomb:
+  write_key: #@ data.values.honeycomb.write_key
+  dataset: #@ data.values.honeycomb.dataset
+#@ end
+
+#@overlay/match by=overlay.subset({"metadata": {"name": "cloud-controller-ng-yaml"}})
+---
+data:
+  #@overlay/replace via=lambda a,_: yaml.encode(overlay.apply(yaml.decode(a), updates()))
+  cloud_controller_ng.yml:
```

Comment:
I think the overlay is overkill here. could we do this with default-empty keys in values.yml?

Reply:
Which part of the overlay?

I ended up taking this route because my initial attempts ran aground on `ytt`'s insistence that any values that you want to configure be present in the first `data` file it gets passed -- which, in this case, is `values/00-values.yml` from cf-for-k8s, which we don't control.

They're reluctant to take a PR adding these values, as it's a relatively niche use case, so this is intended as a transitional step to demonstrate value. (And I might end up swapping this out for an OpenTelemetry-based solution anyway, which makes this much more widely useful, but this is what I currently have working.)

Reply:
> ytt's insistence that any values that you want to configure be present in the first data file it gets passed -- which, in this case, is values/00-values.yml from cf-for-k8s, which we don't control.

I think by using https://github.com/k14s/ytt/blob/develop/docs/ytt-data-values.md#library-data-values , you can have them in the capi-k8s-release.yml values file, and then change them from outside